modules/registry/registry-core/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/JobStatusRepository.java

```diff
@@ -113,4 +117,51 @@ public List<JobStatus> getDistinctListofJobStatus(String gatewayId,String status
         return  jobStatusRepository.selectWithNativeQuery(QueryConstants.FIND_JOB_COUNT_NATIVE_QUERY,
                 gatewayId,status,String.valueOf(time));
     }
+
+    public List<CpuUsage> getCpuUsages(String gatewayId, long fromTime, long toTime) {
+        JobStatusRepository jobStatusRepository = new JobStatusRepository();
+        List<CpuUsage> cpuUsages = new ArrayList<>();
+        List<JobStatus> startedJobStatusList = jobStatusRepository.selectWithNativeQuery(
+                QueryConstants.FIND_JOB_STATUS_STARTED_BEFORE_TIME, gatewayId, JobState.ACTIVE.name(),
+                String.valueOf(toTime));
+        ;
+        List<JobStatus> finishedJobStatusList = jobStatusRepository.selectWithNativeQuery(
+                QueryConstants.FIND_JOB_STATUS_FINISHED_AFTER_TIME, gatewayId, JobState.COMPLETE.name(),
+                JobState.FAILED.name(), JobState.CANCELED.name(), String.valueOf(fromTime));
+        Mapper mapper = ObjectMapperSingleton.getInstance();
+        for (JobStatus startedJobStatus : startedJobStatusList) {
```

Comment:
This double for loop isn't going to scale well for gateways with a large number of jobs. It would be better to do this all in the query. I would suggest something like this:

```sql
select j.* from JOB j
inner join PROCESS p on p.PROCESS_ID = j.PROCESS_ID
inner join EXPERIMENT e on e.EXPERIMENT_ID = p.EXPERIMENT_ID
where e.GATEWAY_ID = ?
and exists (select 1 from JOB_STATUS js where js.JOB_ID = j.JOB_ID and js.STATE = ? and js.TIME_OF_STATE_CHANGE <= ?)
and exists (select 1 from JOB_STATUS js where js.JOB_ID = j.JOB_ID and (js.STATE = ? or js.STATE = ? or js.STATE = ?) and js.TIME_OF_STATE_CHANGE >= ?)
```